src/main/java/sx/blah/discord/handle/obj/ICategory.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * Discord4J is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Discord4J is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Discord4J.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package sx.blah.discord.handle.obj;
+
+import sx.blah.discord.util.cache.LongMap;
+
+import java.util.EnumSet;
+import java.util.List;
+
+/**
+ * A category in Discord.
+ */
+public interface ICategory extends IDiscordObject<ICategory> {
+
+	/**
+	 * Deletes the category.
+	 */
+	void delete();
+
+	/**
+	 * Gets whether the category is deleted.
+	 *
+	 * @return Whether the category is deleted.
+	 */
+	boolean isDeleted();
+
+	/**
+	 * Gets the category's channels.
+	 *
+	 * @return The category's channels.
+	 */
+	List<IChannel> getChannels();
+
+	/**
+	 * Gets the parent guild of the category.
+	 *
+	 * @return The parent guild of the category.
+	 */
+	IGuild getGuild();
+
+	/**
+	 * Gets the position of the category in the channel list.
+	 *
+	 * @return The position of the category in the channel list.
+	 */
+	int getPosition();
+
+	/**
+	 * Changes the position of the category.
+	 *
+	 * @param position The position of the category.
+	 */
+	void changePosition(int position);
+
+	/**
+	 * Gets the name of the category.
+	 *
+	 * @return The name of the category.
+	 */
+	String getName();
+
+	/**
+	 * Changes the name of the category.
+	 *
+	 * @param name The name of the category.
+	 */
+	void changeName(String name);
+
+	/**
+	 * Gets whether the category is marked as NSFW (Not Safe For Work).
+	 *
+	 * @return Whether the category is marked as NSFW.
+	 */
+	boolean isNSFW();
+
+	/**
+	 * Changes the nsfw state of the category.
+	 *
+	 * @param isNSFW The new nsfw state of the category.
+	 */
+	void changeNSFW(boolean isNSFW);
+
+	/**
+	 * Gets the permissions a user has in the category, taking into account user and role overrides.
+	 *
+	 * @param user The user to get permissions for.
+	 * @return The permissions the user has in the category.
+	 */
+	EnumSet<Permissions> getModifiedPermissions(IUser user);
+
+	/**
+	 * Gets the permissions a role has in the category, taking into account role overrides.
+	 *
+	 * @param role The role to get permissions for.
+	 * @return The permissions the role has in the category.
+	 */
+	EnumSet<Permissions> getModifiedPermissions(IRole role);
+
+	/**
+	 * Gets the permissions overrides for users. (Key = User ID)
+	 *
+	 * @return The user permissions overrides for the category.
+	 */
+	LongMap<PermissionOverride> getUserOverridesLong();
```

Comment:
Imo, the `Long` part of the name is unnecessary, even if it is consistent with the channel methods.

Reply:
I think it should stay until we rename the methods in channel, or merge this without Long and rename the existing methods in a commit after

Reply:
Need to do the deprecation dance for that. Not opposed, but we should also see if there's anything else we can do that for.